### ISSUE TITLE
Version 1.0.0 appears to be broken with latest version of sbt

### ISSUE THREAD

**Issue by jedesah** [scoverage/sbt-coveralls#62]
Looking at `CoberturaReader` in version 1.0.0, it looks very fragile and sure enough with sbt 0.13.9, source files are not found because they are missing the `src / main / scala` in the path.

I don't know if sbt changed it's behavior in a recent version, I suspect it used to store the class files under target under `src / main / scala` and does not do so anymore. `CoberturaMultiSourceReader` seems like it should behave fine for our build, so maybe all that is needed is a new release.

**Comment by rorygraves** [comment]
I think this needs some more investigation - I had a brief fiddle with this and it was still not working correctly for the multi-module ensime project.  I'm hoping to look at this in anger within the next couple of weeks because I should have some more time in about a week.

**Comment by jedesah** [comment]
As I hinted in the description, I think this is probably fixed in master. If I can confirm this, can we do a snapshot release?

**Comment by rorygraves** [comment]
Yes. Or even 1.0.1

**Comment by jedesah** [comment]
Ok, I just tried the master version of the plugin on our build and it works!

**Comment by alexflav23** [comment]
@jedesah @rorygraves Any chance a new release will be out anytime soon to fix the multi-build issues? It's been quite a while since coverage is missing from the scene and it would be great to restore things to the right order. Let me know if I can help with any dev effort.

**Comment by lenzenc** [comment]
Any update on a fix for this issue?

**Comment by kshakir** [comment]
I just tried a custom build of c64417d, and still ran into this issue on travis. The `CoberturaMultiSourceReader` cannot locate the files, and is generating a coveralls.json with an empty `"source_files": []`. This is also testable locally, by running `sbt coveralls`, where the coveralls.json is still generated, even though it won't be sent to coveralls.io unless further configured.

No idea if this issue is because of a newer version of `sbt`, a non-multi-build problem, or just a general bug in the `CoberturaMultiSourceReader`.

The cobertura.xml generated by `sbt coverage test` lists the paths as:

-  `src/main/scala/org/example/MyClass.scala`

Meanwhile, the `(sourceDirectories in Compile)` [returns](https://github.com/scoverage/sbt-coveralls/blob/c64417dfcf93f8d81a4776912ac607456f1c1270/src/main/scala/org/scoverage/coveralls/CoverallsPlugin.scala#L87):

- `/Users/kshakir/src/example/src/main/scala-2.11`
- `/Users/kshakir/src/example/src/main/scala`
- `/Users/kshakir/src/example/src/main/java`
- `/Users/kshakir/src/example/target/scala-2.11/src_managed/main`

Thus the full paths incorrectly [resolve](https://github.com/scoverage/sbt-coveralls/blob/c64417dfcf93f8d81a4776912ac607456f1c1270/src/main/scala/org/scoverage/coveralls/CoberturaMultiSourceReader.scala#L47) to:

- `/Users/kshakir/src/example/src/main/scala/src/main/scala/org/example/MyClass.scala`

A workaround that works for me in `.travis.yml` is to hack the sources used in the `after_success`:

```
sudo: false
language: scala
scala:
  - 2.11.7
jdk:
  - oraclejdk8

script: sbt clean coverage test
after_success: sbt 'set unmanagedSourceDirectories in Compile := Seq(file("."))' coveralls
```

**TL;DR: Try `sbt 'set unmanagedSourceDirectories in Compile := Seq(file("."))' coveralls`**

**Comment by rorygraves** [comment]
Hi, this is a known issue and is on my list to fix as soon as I'm able to. 

Thanks for the hint btw ;)

**Comment by alexflav23** [comment]
Hi @rorygraves,

Any updates?

**Comment by rorygraves** [comment]
I'm working on it this week.

**Comment by jedesah** [comment]
Yay! Just flattred the repo to show my appreciation! :smile:

**Comment by rorygraves** [comment]
I haven't forgotten - I can see the issue, just trying to work out the 'right' fix without borking everything

**Comment by alexflav23** [comment]
@rorygraves Any news on this and anything the community can do to help? This is the main coveralls plugin and its currently broken for the whole eco-system.

**Comment by rorygraves** [comment]
Hi @alexflav23 Sorry for the slow process, life and illness have been slowing me down.  I believe I understand the issue and am hoping to have a PR together this weekend.

**Comment by rorygraves** [comment]
A quick update, I believe I actually have a working fix, I'm just tidying up and PR will be tonight or tomorrow.

**Comment by alexflav23** [comment]
Hi @rorygraves,

That's great news, really looking forward to it.

**Comment by alexflav23** [comment]
Hi @rorygraves,

Any news?

**Comment by rorygraves** [comment]
Waiting on @gslowikowski to review and @sksamuel to merge release.

**Comment by sksamuel** [comment]
@rorygraves are you just wanting me to do a 1.0.1 release of master ?

**Comment by sksamuel** [comment]
I've released 1.0.2 to maven central. Note: That's maven central NOT bintray (as I continue to have issues with that). @alexflav23

**Comment by alexflav23** [comment]
HI @sksamuel,

What's the issue with Bintray, can I help with anything? Otherwise many thanks, means everyone can already start using it.

**Comment by sksamuel** [comment]
Can't get bintray to sync with the sbt repo. But it doesn't matter, there's
no reason to use it. Can just use sonatype which I'm familiar with and
everyone has access to maven central.

**Comment by gslowikowski** [comment]
Update README.md file with latest `sbt-scoverage` and `sbt-coveralls` plugin versions.

**Comment by sksamuel** [comment]
done
